src/topic/TopicMessageQuery.js

```diff
@@ -191,83 +322,109 @@ export default class TopicMessageQuery {
                 this._startTime != null ? this._startTime._toProtobuf() : null,
             consensusEndTime:
                 this._endTime != null ? this._endTime._toProtobuf() : null,
-            limit: this._limit != null ? this._limit : null,
+            limit: this._limit != null ? this._limit.sub(this._counter) : null,
```

Comment:
This does not update the starttime from the last message received for retries. It does not even store the last message.

Reply:
You're right, this should keep track of `startTime` of the last message, but I don't see why we should keep the last message completely. From what I can tell all that matters is the `startTime` and making sure we decrement `limit` when needed.

Reply:
True, you only need consensus start time